### PR TITLE
[Enhancement] Check status precise_code instead of construct OLAPInternalError

### LINKED ISSUE #9513: [Enhancement] Confusing log messages from some if branches when check `Status`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, `OLAPInternalError` will catch the exception stack, however in some cases these stack messages can be confusing:
image
While running the above code, even if the status returned is ok, `OLAPInternalError` is constructed in if condition, and it will capture the stack info. What's worse, Status::assemble_state() will print a WARNING log, though there is no exception `Status`:
image

### Solution

Check Status::precise_code instead of constructing `OLAPInternalError` in if condition.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by platoneko**
# Proposed changes

Issue Number: close #9513

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No
3. Has document been added or modified: No
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I will hold on this PR because I am refactoring status code, And I will unify TStatusCode and ErrorCode. precise_code() in status.h will be removed.

File: be/src/olap/utils.h [hunk 1]
```diff
@@ -218,13 +218,10 @@ class Errno {
 };
 
 inline bool is_io_error(Status status) {
```
Comment (yiguolei) [inline be/src/olap/utils.h:220]: You'd better move this method to common/status.h

Comment (platoneko) [reply be/src/olap/utils.h:220]: LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**